crates/derive/src/metrics/noop.rs

```diff
@@ -0,0 +1,28 @@
+use crate::{metrics::PipelineMetrics, pipeline::StepResult, traits::DerivationPipelineMetrics};
+use alloc::sync::Arc;
+
+impl PipelineMetrics {
+    /// No-op implementation for `PipelineMetrics`.
+    pub fn no_op() -> Self {
+        Self {
+            derivation_pipeline_metrics: Arc::new(NoopDerivationPipelineMetrics),
+            // todo: add more metrics here for each stage
```

Comment:
Preference to create a sub-ticket in github issues over leaving TODO comments in the codebase.

Reply:
I'll finish the no-op metrics for all stages in this PR once you've approved the abstractions schema